### PR TITLE
feat: Update FastAPI endpoint to support nodeSources

### PR DESCRIPTION
**Pull request by leehuwuj**

### CODE REVIEW

File: templates/types/streaming/fastapi/app/api/routers/chat.py [hunk 1]
```diff
@@ -19,8 +23,78 @@ class _ChatData(BaseModel):
...
+    def from_source_nodes(cls, source_nodes: List[NodeWithScore]):
+        return [cls.from_source_node(node) for node in source_nodes]
+
+
 class _Result(BaseModel):
     result: _Message
+    nodes: List[_SourceNodes]
+
+
+class VercelStreamResponse(StreamingResponse):
```
Comment (marcusschiesser) [inline templates/types/streaming/fastapi/app/api/routers/chat.py:49]: better move this to a own file

Comment (marcusschiesser) [reply templates/types/streaming/fastapi/app/api/routers/chat.py:49]: probably then `_SourceNodes` too otherwise you'll have a circular import

Comment (leehuwuj) [reply templates/types/streaming/fastapi/app/api/routers/chat.py]: Thank you, I updated by moving the VercelStreamResponse to a new file but still keeping the _SourceNodes and event_generator in this file. Please let me know if there are any issues.

File: templates/types/streaming/fastapi/app/api/routers/vercel_response.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import json
+from typing import Any
+from fastapi.responses import StreamingResponse
+
+
+class VercelStreamResponse(StreamingResponse):
+    """
+    Class to convert the response from the chat engine to the streaming format expected by Vercel
```
Comment (marcusschiesser) [inline templates/types/streaming/fastapi/app/api/routers/vercel_response.py:8]:
```suggestion
    Class to convert the response from the chat engine to the streaming format expected by Vercel/AI
```

**Review by marcusschiesser — APPROVED** [review]

**Pull request merged**